libp2p/muxers/yamux/yamux.nim

```diff
@@ -88,20 +88,21 @@ proc `$`(header: YamuxHeader): string =
     ) & "}, " & "streamId: " & $header.streamId & ", " & "length: " & $header.length &
     "}"
 
-proc encode(header: YamuxHeader): array[12, byte] =
-  result[0] = header.version
-  result[1] = uint8(header.msgType)
-  result[2 .. 3] = toBytesBE(uint16(cast[uint8](header.flags)))
+proc encode(header: YamuxHeader): seq[byte] =
```

Comment:
not sure what this is about, but an array is stack-allocated and fixed-size while a seq is heap-allocated and variable-size - ie all in all, this is a 10x memory usage regression + associated allocation and gc costs resulting in 100x cpu overhead at least ;) 

the c/assembly code inspector and the profiler is your friend here..